typing/typecore.ml

```diff
@@ -983,7 +983,24 @@ module Constructor = NameChoice (struct
   let get_name cstr = cstr.cstr_name
   let get_type cstr = cstr.cstr_res
   let lookup_all_from_type loc usage path env =
-    Env.lookup_all_constructors_from_type ~loc usage path env
+    match Env.lookup_all_constructors_from_type ~loc usage path env with
+    | [] ->
+        begin match (Env.find_type path env).type_kind with
+        | Type_open ->
+            (* Extension constructors cannot be found by looking at the type
+               declaration.
+               We scan the whole environment to get an accurate spellchecking
+               hint in the subsequent error message *)
+            let filter lbl =
+              let get_type_path cstr = match (repr (get_type cstr)).desc with
+                | Tconstr(p, _,_ ) -> p
+                | _ -> assert false in
```

Comment:
This function is already defined as `get_type_path` (same name) in the body of the NameChoice function. I would propose to remove the duplication by defining, at the toplevel

```ocaml
let get_path typ = match (repr ty).desc with
  | Tconstr(p, _,_ ) -> p
  | _ -> assert false
```

and using `get_path (get_type foo)` at the callsites.

Reply:
Done in a slightly altered way.